oracle/lifecycle.go

```diff
@@ -97,19 +118,18 @@ func (o *ProviderOrchestrator) execProviderFn(
 	o.logger.Error("provider exited", zap.String("provider", p.Name()), zap.Error(err))
 }
 
-// setMainCtx sets the main context for the provider orchestrator.
-func (o *ProviderOrchestrator) setMainCtx(ctx context.Context) (context.Context, context.CancelFunc) {
+// getMainCtx returns the main context for the oracle.
+func (o *OracleImpl) getMainCtx() (context.Context, context.CancelFunc) {
 	o.mut.Lock()
 	defer o.mut.Unlock()
 
-	o.mainCtx, o.mainCancel = context.WithCancel(ctx)
 	return o.mainCtx, o.mainCancel
 }
 
-// getMainCtx returns the main context for the provider orchestrator.
-func (o *ProviderOrchestrator) getMainCtx() (context.Context, context.CancelFunc) {
+func (o *OracleImpl) setMainCtx(ctx context.Context) (context.Context, context.CancelFunc) {
```

Comment:
nit: can we add back the godoc here.

Reply:
[readd comment to setMainCtx](https://github.com/skip-mev/slinky/pull/503/commits/ff9d0f17e656ad9686558b132e9c00bb093771c5)